rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,503 @@
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: linkerd-{{.Values.global.namespace}}-{{.Values.alertmanager.name}}
+subjects:
+- kind: ServiceAccount
+  name: linkerd-{{.Values.alertmanager.name}}
+  namespace: {{.Values.global.namespace}}
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: linkerd-{{.Values.alertmanager.name}}
+  namespace: {{.Values.global.namespace}}
+  labels:
+    {{.Values.global.controllerComponentLabel}}: {{.Values.alertmanager.name}}
+    {{.Values.global.controllerNamespaceLabel}}: {{.Values.global.namespace}}
+```
+
+---
+- ## Service Profile Integration 
+
+Service Profiles is an extremely useful *Linkerd* Feature that enables a developer to get metrics on a per route basis. A Service profile can be generated by *Linkerd* using the `linkerd profile` command and passed onto `kubectl apply`. 
+
+`linkerd profile` can obtain route information in 4 different ways:
+- Swagger
+- Protobuf
+- Auto-creation
+- Template
+
+Hence based on the availability of either of these specifications, we can derive per route metrics and create default Prometheus rules for the same. The `config.yaml` file will allow the specification of either a swagger file or a protobuf file. 
+```yaml
+publicWebhook: false
+alertmanager:
+  config: path/to/config/file.yaml 
+
+prometheus:
+  rules: path/to/rules/file.yaml
+  swaggerSpecPath: path/to/swagger/spec
+  protobufSpecPath: path/to/protobuf/spec
+```
+In the event that both of these are absent, we will revert to the [Auto-Creation](https://linkerd.io/2/tasks/setting-up-service-profiles/#auto-creation) method where we watch live traffic and generate a service profile from the `tap` data.
+
+The information from the Service Profile will allow us to define per route metris and thus per route alerts can be configured out of the box.
+
+As a starting point, this can easily be tested using *Linkerd's* own test apps [Emojivoto](https://github.com/BuoyantIO/emojivoto) and [BooksApp](https://github.com/BuoyantIO/booksapp). 
+
+---
+- ## Linkerd Check
+The `linkerd check` command currently has a check to ensure that Prometheus can talk to the control Plane
+```sh
+√ [prometheus] control plane can talk to Prometheus
+```
+A similar check can be added to ensure that the Alertmanager Addon is functional and working as expected. The check will ensure two things
+- Control Plane can talk to Alertmanager.
+- Prometheus can talk to Alertmanager.
+
+```sh
+√ [alertmanager] control plane can talk to Alertmanager
+√ [alertmanager] Prometheus can talk to Alertmanager
+```
+
+# Deliverables
+- Ability to [install](#providing-an-option-to-include-AlertManager-in-the-installation) Alertmanager as an addon as part of the Control plane. 
+- [Default](#default-rules) out-of-the-box alerts. 
+- [Checks](#linkerd-check), to see that Alertmanager has been successfully installed and configured correctly
+- Simple configuration of [alert channels](#alertmanager-helm-chart) (slack, email).
```

Comment:
Users can do this themselves, as long as we provide an easy way for them to express and pass in the configuration.